plugins/modules/system/sap_task_list_execute.py

```diff
@@ -0,0 +1,437 @@
+
+def run_module():
+    # define available arguments/parameters a user can pass to the module
+    module = AnsibleModule(
+        argument_spec=dict(
+            state=dict(default='present', choices=['show', 'present']),
+            # values for connection
+            conn_username=dict(type='str', required=True),
+            conn_password=dict(type='str', required=True, no_log=True),
+            host=dict(type='str', required=True),
+            sysnr=dict(type='str', default="00"),
+            client=dict(type='str', default="000"),
+            # values for execution tasks
+            task_to_execute=dict(type='str', required=True),
+            task_parameters=dict(type='list', elements='list', required=False),
+            task_settings=dict(type='list', elements='str', default=['BATCH']),
+            tasks_skip=dict(type='bool', default=False),
+        ),
+        supports_check_mode=False,
+    )
+    result = dict(changed=False, msg='', results=[], out={}, stdout={}, stderr={}, )
+
+    params = module.params
+
+    state = params['state']
+
+    username = (params['conn_username']).upper()
+    password = params['conn_password']
+    host = params['host']
+    sysnr = params['sysnr']
+    client = params['client']
+
+    task_parameters = params['task_parameters']
+    task_to_execute = params['task_to_execute']
+    task_settings = params['task_settings']
+    tasks_skip = params['tasks_skip']
+
+    if not HAS_ANOTHER_LIBRARY:
+        module.fail_json(
+            msg=missing_required_lib('pyrfc'),
+            exception=ANOTHER_LIBRARY_IMPORT_ERROR)
+
+    # basic RFC connection with pyrfc
+    connection_params = {'user': username, 'passwd': password, 'ashost': host, 'sysnr': sysnr, 'client': client}
+    try:
+        conn = Connection(**connection_params)
+    except Exception as err:
+        result['stderr'] = str(err)
+        result['msg'] = 'Something went wrong connecting to the SAP system.'
+        module.fail_json(**result)
+
+    if task_to_execute is not None:
+        try:
+            raw_params = call_rfc_method(conn, 'STC_TM_SCENARIO_GET_PARAMETERS', {'I_SCENARIO_ID': task_to_execute})
+        except Exception as err:
+            result['stderr'] = str(err)
+            result['msg'] = 'The task list does not exsist.'
+            module.fail_json(**result)
+
+        if state == "show":
+            params_list = []
+            for tasks in raw_params['ET_PARAM_DEF']:
+                params_list = params_list + [[tasks['TASKNAME'], tasks['FIELDNAME'], tasks['DEFAULTVAL']]]
+            result['results'] = params_list
+            result['stdout'] = json.dumps(raw_params['ET_PARAM_DEF'])
+            result['out'] = raw_params['ET_PARAM_DEF']
+            result['changed'] = True
+
+        if state == "present":
+            exec_settings = process_exec_settings(task_settings)
+            converted_parameters = process_task_parameters(task_parameters)
+
+            if not converted_parameters:
```

Comment:
if task_parameters is `None`, the `process_task_parameters()` function will return `[{'parameter_list': []}, {'task_list': []}]`. When tested in the line above, it will fail (the code will NOT go to 375 and 376 below). Is that what you had in mind?